repositories/local_storage.go

```diff
@@ -30,25 +34,50 @@ func (l LocalStorage) Write(pokemons []models.Pokemon) error {
 	return nil
 }
 
-func (l LocalStorage) Read() ([]models.Pokemon, error) {
+func (l LocalStorage) Read(ctx context.Context) (<-chan models.Pokemon, error) {
 	file, fErr := os.Open(filePath)
-	defer file.Close()
 	if fErr != nil {
 		return nil, fErr
 	}
 
-	r := csv.NewReader(file)
-	records, rErr := r.ReadAll()
-	if rErr != nil {
-		return nil, rErr
-	}
+	return readRecords(file, ctx), nil
```

Comment:
Es [buena practica](https://stackoverflow.com/questions/50522658/why-recommend-use-ctx-as-the-first-parameter) pasar el ctx como el primer argumento a las funciones